source/common/filter/http/filter_config_discovery_impl.cc

```diff
@@ -150,12 +179,22 @@ void FilterConfigSubscription::onConfigUpdate(
     const std::vector<Config::DecodedResourceRef>& added_resources,
     const Protobuf::RepeatedPtrField<std::string>& removed_resources, const std::string&) {
   if (!removed_resources.empty()) {
-    ENVOY_LOG(error,
-              "Server sent a delta ExtensionConfigDS update attempting to remove a resource (name: "
-              "{}). Ignoring.",
-              removed_resources[0]);
-  }
-  if (!added_resources.empty()) {
+    ASSERT(removed_resources.size() == 1);
```

Comment:
This is because we're subscribing to a singleton resource, right?

Reply:
Yeah, we already have similar assumptions in the existing code, e.g. `onConfigUpdate(added_resources, added_resources[0].get().version());`